### ISSUE TITLE
Provide access to AWS Stage Variables

### ISSUE THREAD

**Issue by a-golovanov** [apex/up#236]
KMS is a good solution for storing secrets, but it's not free. During development I would rather use stage variables instead, but apparently there is no way to access them right now.
There are at least two possible solutions that come to my mind:
1. Pass in headers. Either as a single header with JSON or as a bunch of headers like `X-AWS-STAGE-VARIABLE-1: Name=Value` whichever seems more idiomatic (I'm not an expert in this area).
2. Transfer to environment variables before starting lambda server process - this one will require process restart if stage variables changed and lambda is recycled.

This is sort of a side topic for https://github.com/apex/up/issues/90

**Comment by tj** [comment]
👍 . They're a little awkward to pass as env vars, I thought about doing that too, just setting them on the first request, feels pretty hacky. I'm not sure they're really designed for anything other than the templates since they're not encrypted either, and "proper" env vars for Lambda are shipped per-lambda haha. AWS is confused :D

**Comment by a-golovanov** [comment]
I agree that setting stage variables as environment variables on first request is hacky that's why I listed it second and added restart process bit. 

I also agree that stage variables are not the way to store production secrets, lambda environment variables neither. The downsides of lambda environment variables use in development are that they would need to be committed to git and would be present in production builds.

Will you consider pass stage variables as http header field(s)? This seems like a quick solution.

**Comment by a-golovanov** [comment]
I looked closer at AWS Parameter Store. It can be used with and without encryption (i.e. KMS), so I can use it uniformly for development and production. Seems like a I don't need stage variables after all.

**Comment by maxbeatty** [comment]
I'd like to use Stage Variables to set `Access-Control-Allow-Origin` headers for CORS requests
